.travis.yml

```diff
@@ -7,7 +7,7 @@ jdk:
 
 cache:
   directories:
-  - "$HOME/.m2"
+  - "$HOME/.m2/repository"
```

Comment:
I don't understand to change. To me it is still caching but for a different directory. What is the `Maven configuration` in the description? Where are they? Can you explain more?

Reply:
[~/.m2/settings.xml](https://maven.apache.org/settings.html) is some Maven configuration. Normally .m2 would be empty when we start running, so it wouldn't be saved. But Travis actually pre-loads the .m2 directory with a custom settings.xml (which has [bitten us before](https://github.com/grpc/grpc-java/commit/3b578abf998eb59cc494bd850385b796239ee43e)).

What happened is the pre-loaded settings.xml then got included in our build cache, and then we started using an out-dated settings.xml from our cache instead of any newer pre-loaded settings.xml that Travis provides. I could have just cleared the Travis cache, but then this sort of problem could reoccur in the future.